src/core/ucc_progress_queue_mt.c

```diff
@@ -100,6 +101,20 @@ static int ucc_pq_mt_progress(ucc_progress_queue_t *pq)
     return n_progressed;
 }
 
+static int ucc_pq_locked_mt_is_empty(ucc_progress_queue_t *pq)
+{
+    ucc_pq_mt_locked_t *pq_mt = ucc_derived_of(pq, ucc_pq_mt_locked_t);
+
+/* this function should not be very accurate for the purpose of progress throttling */
```

Comment:
I am not sure to understand the comment "should not be very accurate"

Also need to fix indentation. And maybe prefer inline comment style with "//" ?

Reply:
There is a race condition here which might result in returning false result from is_empty(). To avoid race condition we need to use locks, however for performance it's better to call progress few extra times comparing to locking/unlocking  every single is_empty() call.

Reply:
got it! Interesting